runtime/j9vm/jvm.c

```diff
@@ -3589,13 +3589,24 @@ JVM_LoadLibrary(const char* libName)
 	J9JavaVM* javaVM = (J9JavaVM*)BFUjavaVM;
 	J9InternalVMFunctions* vmFuncs = javaVM->internalVMFunctions;
 	J9VMThread* currentThread = vmFuncs->currentVMThread(javaVM);
+	UDATA flagVMAccess = 0;
 
-	Trc_SC_LoadLibrary_Entry(libName);
+	Trc_SC_LoadLibrary_Entry(currentThread, libName);
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+	flagVMAccess = currentThread->publicFlags & J9_PUBLIC_FLAGS_VM_ACCESS;
+	Trc_SC_LoadLibrary_FlagVMAccess(currentThread, flagVMAccess);
+	if (0 != flagVMAccess) {
+		vmFuncs->internalReleaseVMAccess(currentThread);
+	}
 	if (vmFuncs->registerBootstrapLibrary(currentThread, libName, &nativeLibrary, FALSE) == J9NATIVELIB_LOAD_OK) {
 		result = (void*)nativeLibrary->handle;
 	}
+	if (0 != flagVMAccess) {
```

Comment:
No need to acquire here - exit will work regardless.

Reply:
Removed the VM access acquiring.

Please have another look.